lib/semian/mysql2.rb

```diff
@@ -57,6 +61,17 @@ def semian_identifier
       end
     end
 
+    def ping
+      result = nil
+      acquire_semian_resource(adapter: :mysql, scope: :ping) do
+        result = raw_ping
+        raise PingFailure.new(result.to_s) unless result
+      end
+      result
+    rescue ResourceBusyError, CircuitOpenError, PingFailure
+      false
```

Comment:
Since establishing a new connection is also behind a circuit, do you think this is problematic? In vanilla AR, what would happen is that the connection is checked out of the pool, on `verify!` `#ping` returns false and a new connection is established. This would raise right away. Given that Rails doesn't check out the connection by default anymore in 4.2 (only when it's required), this seems sensible.

Internally, we only reconnect on `#execute` (.. and `#quote`), and we'd only do so when a `#ping` fails (when it's circuit broken), in which case it'd often just raise straight away.

I can't think of a case where things would go nuts in AR, and given that this doesn't change behaviour but just enforces it more strictly and faster during certain conditions, I'm not super concerned. Is there anything in particular you'd like me to investigate?